document_parameters.rb

```diff
@@ -1,33 +1,44 @@
-require_relative 'rosette_api_error'
+require_relative 'bad_request_format_error'
 
 class DocumentParameters
   attr_accessor :content, :content_uri, :file_path, :language
 
-  def initialize(content=nil, content_uri=nil, file_path=nil, language=nil)
-    @content = content
-    @content_uri = content_uri
-    @file_path = file_path
-    @language = language
+  def initialize(options = {})
+    options = {
+      'content' => nil,
+      'content_uri' => nil,
+      'file_path' => nil,
+      'language' => nil
+    }.update options
+    @content = options['content']
+    @content_uri = options['content_uri']
+    @file_path = options['file_path']
+    @language = options['language']
   end
 
   def validate_params
-    if [@content, @content_uri, @file_path].count { |attr| !attr.nil? } > 1
-      raise RosetteAPIError.new('badRequestFormat', 'The format of the request is invalid: multiple content sources;' \
-          ' must be one of an attachment, an inline "content" field, or an external "contentUri"')
-    elsif [@content, @content_uri, @file_path].count { |attr| attr.nil? } == 3
-      raise RosetteAPIError.new('badRequestFormat', 'The format of the request is invalid: no content provided; must' \
-          ' be one of an attachment, an inline "content" field, or an external "contentUri"')
+    if [@content, @content_uri, @file_path].compact.length > 1
+      raise BadRequestFormatError.new 'The format of the request is invalid: multiple content sources;' \
+                                      ' must be one of an attachment, an inline "content" field, or an external "contentUri"'
+    elsif [@content, @content_uri, @file_path].all?(&:nil?)
+      raise BadRequestFormatError.new 'The format of the request is invalid: no content provided; must' \
+                                      ' be one of an attachment, an inline "content" field, or an external "contentUri"'
     end
   end
 
   def load_params
-    validate_params
-    self.to_hash.select { |k, v| !v.nil? }.map { |k, v| [k.to_s.split('_').collect(&:capitalize).join.sub!(/\D/, &:downcase), v] }.to_h
+    self.validate_params
+    self.to_hash.select { |key, value| !value.nil? }
+                .map { |key, value| [key.to_s.split('_').map(&:capitalize).join.sub!(/\D/, &:downcase), value] }
+                .to_h
   end
 
   def to_hash
-    hash = {}
-    instance_variables.each { |var| hash[var.to_s.delete('@')] = instance_variable_get(var) }
-    hash
+    {
```

Comment:
This is slightly more code but also a much clearer way of creating the hash. I almost never have any need for `#instance_variables` and the related getters and setters.